eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/VariantExporter.java

```diff
@@ -65,43 +64,91 @@
      */
     public VariantExporter(CellbaseWSClient cellbaseClient) {
         this.cellbaseClient = cellbaseClient;
+        outputSampleNames = new HashSet<>();
     }
 
-    public Map<String, List<VariantContext>> export(VariantDBIterator iterator, Region region, List<String> studyIds) {
-        Map<String, List<VariantContext>> variantsToExportByStudy = new HashMap<>();
+    public List<VariantContext> export(VariantDBIterator iterator, Region region) {
+        List<VariantContext> variantsToExport = new ArrayList<>();
         failedVariants = 0;
 
         // region sequence contains the last exported region: we set it to null to get the new region sequence from cellbase if needed
-        regionSequence = null;
+        variantToVariantContextConverter.cleanCachedRegionSequence();
 
         while (iterator.hasNext()) {
             Variant variant = iterator.next();
             if (region.contains(variant.getChromosome(), variant.getStart())) {
                 try {
-                    Map<String, VariantContext> variantContexts = convertBiodataVariantToVariantContext(variant, studyIds, region);
-                    for (Map.Entry<String, VariantContext> variantContext : variantContexts.entrySet()) {
-                        variantsToExportByStudy.putIfAbsent(variantContext.getKey(), new ArrayList<>());
-                        variantsToExportByStudy.get(variantContext.getKey()).add(variantContext.getValue());
-                    }
+                    VariantContext variantContext = variantToVariantContextConverter.transform(variant, region);
+                    variantsToExport.add(variantContext);
                 } catch (Exception e) {
                     logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(), variant.getReference(),
                             variant.getAlternate(), e.getMessage());
                     failedVariants++;
                 }
             }
         }
-        return variantsToExportByStudy;
+        return variantsToExport;
     }
 
-    public Map<String, VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor, List<String> studyIds) {
+    public Map<String, VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor, List<String> studyIds) throws IllegalArgumentException {
         Map<String, VariantSource> sources = new TreeMap<>();
         List<VariantSource> sourcesList = sourceDBAdaptor.getAllSourcesByStudyIds(studyIds, new QueryOptions()).getResult();
+        checkIfTereAreSourceForEveryStudy(studyIds, sourcesList);
+        Map<String, Map<String, String>> studiesSampleNamesMapping = checkIfConflictsInSampleNames(sourcesList);
+        variantToVariantContextConverter = new BiodataVariantToVariantContextConverter(sourcesList, cellbaseClient, studiesSampleNamesMapping);
```

Comment:
Even if convenient due to variables scope, I'm not sure this initialization belongs in this method.